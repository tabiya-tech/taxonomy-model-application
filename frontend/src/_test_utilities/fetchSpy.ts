export function setupFetchSpy(
  expectedStatus: number,
  expectedResponseBody: string | object | undefined,
  contentType: "" | "application/json;charset=UTF-8"
): jest.SpyInstance {
  const responseBody =
    typeof expectedResponseBody === "string" ? expectedResponseBody : JSON.stringify(expectedResponseBody);
  const expectedResponse = new Response(responseBody, {
    status: expectedStatus,
    headers: { "Content-Type": contentType },
  });
  return jest.spyOn(window, "fetch").mockResolvedValue(expectedResponse);
}
