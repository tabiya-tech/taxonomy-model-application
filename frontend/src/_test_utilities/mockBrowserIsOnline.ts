import { fireEvent } from "@testing-library/react";

export function mockBrowserIsOnLine(isOnline: boolean) {
  if (window.navigator.onLine !== isOnline) {
    jest.spyOn(window.navigator, "onLine", "get").mockReturnValue(isOnline);
    if (isOnline) {
      fireEvent.online(window);
    } else {
      fireEvent.offline(window);
    }
  }
}

export function unmockBrowserIsOnLine() {
  jest.spyOn(window.navigator, "onLine", "get").mockRestore();
}
