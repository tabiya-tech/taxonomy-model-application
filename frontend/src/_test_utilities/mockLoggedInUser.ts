import { AuthContextValue, TabiyaUser, TabiyaUserRole, authContextDefaultValue } from "src/auth/AuthProvider";

export function mockLoggedInUser(mockedValues: Partial<AuthContextValue>){
  let user: TabiyaUser = mockedValues.user || { username: "foo", roles: [] };
  jest.spyOn(require("src/auth/hooks/useAuthUser"), "useAuthUser").mockReturnValue({
    ...authContextDefaultValue,
    user: user,
    setUser: (user: TabiyaUser) => null,
    hasRole: (role: TabiyaUserRole) => user.roles.includes(role),
    updateUserByAccessToken: () => {},
    ...mockedValues
  });
}
