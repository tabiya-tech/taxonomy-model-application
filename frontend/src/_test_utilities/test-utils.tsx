// Based on https://testing-library.com/docs/react-testing-library/setup/

import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/providers";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
