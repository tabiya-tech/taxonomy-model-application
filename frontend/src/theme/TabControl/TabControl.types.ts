import { ReactNode } from "react";

/**
 *  TabControlConfig takes an id, label, and panel
 *  @param id - the id of the tab, should be a unique string for each tab, used as list keys
 *  @param label - the label of the tab, the tab name
 *  @param panel - the panel of the tab, a react component to be rendered when the tab is active
 */
export type TabControlConfig = {
  id: string;
  label: string;
  panel?: ReactNode;
};
