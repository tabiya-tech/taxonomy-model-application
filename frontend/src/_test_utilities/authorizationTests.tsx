import { render, screen } from "src/_test_utilities/test-utils";
import AuthAPISpecs from "api-specifications/auth";
import { mockLoggedInUser } from "./mockLoggedInUser";

export const ALL_USERS = [
  AuthAPISpecs.Enums.TabiyaRoles.ANONYMOUS,
  AuthAPISpecs.Enums.TabiyaRoles.REGISTERED_USER,
  AuthAPISpecs.Enums.TabiyaRoles.MODEL_MANAGER,
];

type AuthorizationTestsParams = {
  name: string;
  Component: JSX.Element;
  testIds?: string[];
  roles?: AuthAPISpecs.Enums.TabiyaRoles[];
  tests?: Partial<
    Record<
      AuthAPISpecs.Enums.TabiyaRoles,
      {
        visible?: string[];
        hidden?: string[];
        visibleDisabled?: string[];
      }
    >
  >;
};

export const authorizationTests = {
  defaultName: "Test the rendering of the component based on Roles",
  callback: ({ name, Component, testIds = [], roles = ALL_USERS, tests }: AuthorizationTestsParams) => {
    return () => {
      test.each(roles?.map((role) => [role]))(`${name} renders correctly for user with role %s`, (role) => {
        const commonElements = testIds || [];
        const { visible = [], visibleDisabled = [], hidden = [] } = tests?.[role] || {};

        mockLoggedInUser({ user: { username: "foo", roles: [role] } });

        render(Component);

        commonElements.forEach((testId) => expect(screen.getByTestId(testId)).toBeInTheDocument());

        visible?.forEach((testId) => {
          expect(screen.getByTestId(testId)).toBeVisible();
        });

        hidden?.forEach((testId) => {
          expect(screen.queryByTestId(testId)).not.toBeInTheDocument();
        });

        visibleDisabled?.forEach((testId) => {
          const element = screen.getByTestId(testId);
          expect(element).toBeVisible();
          if (element instanceof HTMLInputElement) {
            expect(element).toBeDisabled();
          }
        });
      });
    };
  },
};
