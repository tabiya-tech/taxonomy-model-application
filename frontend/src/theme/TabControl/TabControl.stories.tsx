import { Meta, StoryObj } from "@storybook/react";
import TabControl from "./TabControl";
import { VisualMock } from "src/_test_utilities/VisualMock";
import { getRandomLorem } from "src/_test_utilities/specialCharacters";

const meta: Meta<typeof TabControl> = {
  title: "Components/TabControl",
  component: TabControl,
  tags: ["autodocs"],
  argTypes: {},
};

export default meta;

type Story = StoryObj<typeof TabControl>;

export const Shown: Story = {
  args: {
    items: [
      { id: "1", label: "Tab 1", panel: <VisualMock text={"Tab 1 Content"} /> },
      { id: "2", label: "Tab 2", panel: <VisualMock text={"Tab 2 Content"} /> },
      { id: "3", label: "Tab 3", panel: <VisualMock text={"Tab 3 Content"} /> },
    ],
  },
};

export const ShownWithLongText: Story = {
  args: {
    items: [
      { id: "1", label: getRandomLorem(50), panel: <VisualMock text={getRandomLorem(500)} /> },
      { id: "2", label: getRandomLorem(100), panel: <VisualMock text={getRandomLorem(750)} /> },
      { id: "3", label: getRandomLorem(150), panel: <VisualMock text={getRandomLorem(1000)} /> },
    ],
  },
};
