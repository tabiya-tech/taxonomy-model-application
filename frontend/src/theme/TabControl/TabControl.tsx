import { TabControlConfig } from "./TabControl.types";
import React, { useState } from "react";
import { Box, Tab, Tabs, useTheme } from "@mui/material";
import { motion } from "framer-motion";

interface StyledTabsProps {
  children?: React.ReactNode;
  "aria-label": string;
  value: number;
  onChange: (event: React.SyntheticEvent, newValue: number) => void;
}

const StyledTabs = (props: StyledTabsProps) => (
  <Tabs
    {...props}
    variant="fullWidth"
    selectionFollowsFocus
    sx={{
      backgroundColor: (theme) => theme.palette.common.white,
      position: "sticky",
      top: 0,
      zIndex: 1,
      "& .MuiTabs-indicator": {
        display: "flex",
        justifyContent: "center",
        backgroundColor: "transparent",
      },
      "& .MuiTabs-indicatorSpan": {
        maxWidth: "40%",
        width: "100%",
        backgroundColor: (theme) => theme.palette.primary.main,
      },
    }}
    TabIndicatorProps={{ children: <span className="MuiTabs-indicatorSpan" /> }}
  />
);

export interface TabControlProps {
  "aria-label": string;
  items: TabControlConfig[];
  "data-testid"?: string;
}

/**
 * TabControl responsible for rendering tabs and content given a list of TabControlConfig items
 * @param items a list of TabControlConfig items with an id, label and panel component
 * @param props gets a data-testid for testing from parent
 * @constructor
 */

const uniqueId = "c356cfd6-5a7a-4471-9ef3-4c1932251bcc";
export const DATA_TEST_ID = {
  TAB_CONTROL_LABEL: `tab-control-tabs-${uniqueId}`,
  TAB_CONTROL_PANEL: `tab-control-panel-${uniqueId}`,
};

const TabControl: React.FC<TabControlProps> = (props: Readonly<TabControlProps>) => {
  const [value, setValue] = useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };
  const theme = useTheme();
  return (
    <Box display="flex" flexDirection={"column"} sx={{ height: "100%" }} data-testid={props["data-testid"]}>
      <StyledTabs value={value} onChange={handleChange} aria-label={props["aria-label"]}>
        {props.items.map((item) => (
          <Tab
            sx={{
              ...theme.typography.button,
            }}
            label={item.label}
            key={item.id}
            data-testid={DATA_TEST_ID.TAB_CONTROL_LABEL}
            id={`tab-${item.id}`}
          />
        ))}
      </StyledTabs>
      {props.items.map(
        (item, index) =>
          value === index && (
            <Box
              role="tabpanel"
              id={`tabpanel-${item.id}`}
              aria-labelledby={`tab-${item.id}`}
              data-testid={DATA_TEST_ID.TAB_CONTROL_PANEL}
              key={item.id}
              sx={{
                paddingBottom: theme.spacing(theme.tabiyaSpacing.lg),
                maxHeight: "100%",
                paddingTop: theme.spacing(theme.tabiyaSpacing.lg),
                flexGrow: 1,
                overflowY: "auto",
                overflowX: "hidden",
              }}
            >
              {
                <motion.div initial={{ opacity: 0.2 }} animate={{ opacity: 1 }} transition={{ duration: 1 }}>
                  {item.panel}
                </motion.div>
              }
            </Box>
          )
      )}
    </Box>
  );
};

export default TabControl;
