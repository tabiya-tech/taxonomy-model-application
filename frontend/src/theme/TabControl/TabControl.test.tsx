// mock the chatty console
import "src/_test_utilities/consoleMock";

import TabControl, { DATA_TEST_ID } from "./TabControl";
import { render, screen, within, act, waitFor } from "src/_test_utilities/test-utils";
import { TabControlConfig } from "./TabControl.types";

describe("TabControl component render tests", () => {
  test("renders correctly with no items", () => {
    // GIVEN an empty array of `items`
    // AND a `data-testid`
    const dataTestId = "test-id";
    // WHEN the TabControl component is rendered
    render(<TabControl items={[]} data-testid={dataTestId} aria-label={"Test tabs"} />);
    // THEN the component should be rendered
    const tabControlComponent = screen.getByTestId(dataTestId);
    expect(tabControlComponent).toBeInTheDocument();
    // THEN it is rendered with no labels
    const labelComponent = screen.queryByTestId(DATA_TEST_ID.TAB_CONTROL_LABEL);
    expect(labelComponent).not.toBeInTheDocument();
    // AND it is rendered with no panels
    const panelComponent = screen.queryByTestId(DATA_TEST_ID.TAB_CONTROL_PANEL);
    expect(panelComponent).not.toBeInTheDocument();
    // AND match the snapshot
    expect(tabControlComponent).toMatchSnapshot();
    // AND no errors should be logged to the console
    expect(console.error).not.toHaveBeenCalled();
  });

  test("renders tabs and content for a single provided item", async () => {
    // GIVEN a single `TabControlConfig` item with unique `id`, `label`, and `panel`
    const givenPanelName = "Test Panel";
    const givenItem: TabControlConfig = {
      id: "unique-id",
      label: "Test Label",
      panel: <div>{givenPanelName}</div>,
    };
    // AND a `data-testid`
    const givenTestId = "test-id";
    // WHEN the TabControl component is rendered
    render(<TabControl items={[givenItem]} data-testid={givenTestId} aria-label={"Test tabs"} />);
    // THEN it should display the component
    const tabControlComponent = screen.getByTestId(givenTestId);
    expect(tabControlComponent).toBeInTheDocument();
    // AND it should display one tab with the `label`
    const tabLabel = within(tabControlComponent).getByTestId(DATA_TEST_ID.TAB_CONTROL_LABEL);
    expect(tabLabel).toBeInTheDocument();
    // AND display the panel content
    const tabPanel = within(tabControlComponent).getByTestId(DATA_TEST_ID.TAB_CONTROL_PANEL);
    expect(tabPanel).toBeInTheDocument();
    // AND match the snapshot
    expect(tabControlComponent).toMatchSnapshot();

    // AND WHEN some time has passed
    // THEN the component is rendered visible
    await waitFor(() => expect(screen.getByTestId(givenTestId)).toBeVisible());
    // AND the label is rendered visible
    await waitFor(() => expect(screen.getByTestId(DATA_TEST_ID.TAB_CONTROL_LABEL)).toBeVisible());
    // AND the panel is rendered visible
    await waitFor(() => expect(screen.getByTestId(DATA_TEST_ID.TAB_CONTROL_PANEL)).toBeVisible());

    // AND no errors should be logged to the console
    expect(console.error).not.toHaveBeenCalled();
  });

  test("renders tabs and content for multiple provided items", async () => {
    // GIVEN an array of `TabControlConfig` items with unique `id`, `label`, and `panel` components
    const givenPanelName1 = "Test Panel 1";
    const givenPanelName2 = "Test Panel 2";
    const givenItems: TabControlConfig[] = [
      {
        id: "unique-id-1",
        label: "Test Label 1",
        panel: <div>{givenPanelName1}</div>,
      },
      {
        id: "unique-id-2",
        label: "Test Label 2",
        panel: <div>{givenPanelName2}</div>,
      },
    ];
    // AND a `data-testid`
    const givenTestId = "test-id";
    // WHEN the TabControl component is rendered
    render(<TabControl items={givenItems} data-testid={givenTestId} aria-label={"Test tabs"} />);
    // AND it should display the component
    const tabControlComponent = screen.getByTestId(givenTestId);
    expect(tabControlComponent).toBeInTheDocument();
    // THEN it should display tabs for each item based on the `label`
    expect(screen.getByText(givenItems[0].label)).toBeInTheDocument();
    expect(screen.getByText(givenItems[1].label)).toBeInTheDocument();
    // AND display the first item's panel content by default
    expect(screen.getByText("Test Panel 1")).toBeInTheDocument();
    // AND match the snapshot
    expect(tabControlComponent).toMatchSnapshot();

    // AND after some time has passed
    // THEN the component is rendered visible
    await waitFor(() => expect(screen.getByTestId(givenTestId)).toBeVisible());
    // AND the first label is rendered visible
    await waitFor(() => expect(screen.getByText(givenItems[0].label)).toBeVisible());
    // AND the second label is rendered visible
    await waitFor(() => expect(screen.getByText(givenItems[1].label)).toBeVisible());
    // AND the first panel is rendered visible
    await waitFor(() => expect(screen.getByText(givenPanelName1)).toBeVisible());

    // AND no errors should be logged to the console
    expect(console.error).not.toHaveBeenCalled();
  });
});

describe("TabControl component interaction tests", () => {
  test("switches content when a tab is clicked", async () => {
    // GIVEN an array of `TabControlConfig` items with unique `id`, `label`, and `panel` components
    const givenPanelName1 = "Test Panel 1";
    const givenPanelName2 = "Test Panel 2";
    const givenPanelName3 = "Test Panel 3";
    const givenItems: TabControlConfig[] = [
      {
        id: "unique-id-1",
        label: "Test Label 1",
        panel: <div>{givenPanelName1}</div>,
      },
      {
        id: "unique-id-2",
        label: "Test Label 2",
        panel: <div>{givenPanelName2}</div>,
      },
      {
        id: "unique-id-3",
        label: "Test Label 3",
        panel: <div>{givenPanelName3}</div>,
      },
    ];
    // AND a `data-testid`
    const givenTestId = "test-id";
    // AND the TabControl component is rendered
    render(<TabControl items={givenItems} data-testid={givenTestId} aria-label={"Test tabs"} />);

    // AND it should display the component
    const tabControlComponent = screen.getByTestId(givenTestId);
    expect(tabControlComponent).toBeInTheDocument();
    // AND it should display tabs for each item based on the `label`
    expect(screen.getByText(givenItems[0].label)).toBeInTheDocument();
    expect(screen.getByText(givenItems[1].label)).toBeInTheDocument();
    expect(screen.getByText(givenItems[2].label)).toBeInTheDocument();
    // AND display the first item's panel content by default
    expect(screen.getByText(givenPanelName1)).toBeInTheDocument();

    // WHEN the second tab label is clicked
    act(() => {
      screen.getByText(givenItems[1].label).click();
    });
    // THEN the second item's panel content should be displayed
    expect(await screen.findByText(givenPanelName2)).toBeInTheDocument();
    // AND the first item's panel content should not be displayed
    await waitFor(() => expect(screen.queryByText(givenPanelName1)).not.toBeInTheDocument());
    // AND the third item's panel content should not be displayed
    await waitFor(() => expect(screen.queryByText(givenPanelName3)).not.toBeInTheDocument());

    // AND after some time has passed
    // THEN the second panel is rendered visible
    await waitFor(() => expect(screen.getByText(givenPanelName2)).toBeVisible());

    // AND WHEN the third tab label is clicked
    act(() => {
      screen.getByText(givenItems[2].label).click();
    });
    // THEN the third item's panel content should be displayed
    expect(await screen.findByText(givenPanelName3)).toBeInTheDocument();
    // AND the first item's panel content should not be displayed
    await waitFor(() => expect(screen.queryByText(givenPanelName1)).not.toBeInTheDocument());
    // AND the second item's panel content should not be displayed
    await waitFor(() => expect(screen.queryByText(givenPanelName2)).not.toBeInTheDocument());

    // AND after some time has passed
    // THEN the third panel is rendered visible
    expect(await screen.findByText(givenPanelName3)).toBeVisible();

    // AND WHEN the first tab label is clicked again
    act(() => {
      screen.getByText(givenItems[0].label).click();
    });
    // AND after some time has passed
    // THEN the first item's panel content should be displayed
    expect(await screen.findByText(givenPanelName1)).toBeInTheDocument();
    // AND the second item's panel content should not be displayed
    await waitFor(() => expect(screen.queryByText(givenPanelName2)).not.toBeInTheDocument());
    // AND the third item's panel content should not be displayed
    await waitFor(() => expect(screen.queryByText(givenPanelName3)).not.toBeInTheDocument());

    // THEN the first panel is rendered visible
    expect(await screen.findByText(givenPanelName1)).toBeVisible();

    // AND no errors should be logged to the console
    expect(console.error).not.toHaveBeenCalled();
  });
});
