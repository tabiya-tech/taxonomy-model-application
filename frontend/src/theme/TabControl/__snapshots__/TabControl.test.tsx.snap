// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabControl component render tests renders correctly with no items 1`] = `
<div
  class="MuiBox-root css-1g4yje1"
  data-testid="test-id"
>
  <div
    class="MuiTabs-root css-6gqcb5-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        aria-label="Test tabs"
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      />
      <span
        class="MuiTabs-indicator css-1ktk2h2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      >
        <span
          class="MuiTabs-indicatorSpan"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`TabControl component render tests renders tabs and content for a single provided item 1`] = `
<div
  class="MuiBox-root css-1g4yje1"
  data-testid="test-id"
>
  <div
    class="MuiTabs-root css-6gqcb5-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        aria-label="Test tabs"
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-l3v1a8-MuiButtonBase-root-MuiTab-root"
          data-testid="tab-control-tabs-c356cfd6-5a7a-4471-9ef3-4c1932251bcc"
          id="tab-unique-id"
          role="tab"
          tabindex="0"
          type="button"
        >
          Test Label
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="MuiTabs-indicator css-1ktk2h2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      >
        <span
          class="MuiTabs-indicatorSpan"
        />
      </span>
    </div>
  </div>
  <div
    aria-labelledby="tab-unique-id"
    class="MuiBox-root css-1j1uas5"
    data-testid="tab-control-panel-c356cfd6-5a7a-4471-9ef3-4c1932251bcc"
    id="tabpanel-unique-id"
    role="tabpanel"
  >
    <div
      style="opacity: 0.2;"
    >
      <div>
        Test Panel
      </div>
    </div>
  </div>
</div>
`;

exports[`TabControl component render tests renders tabs and content for multiple provided items 1`] = `
<div
  class="MuiBox-root css-1g4yje1"
  data-testid="test-id"
>
  <div
    class="MuiTabs-root css-6gqcb5-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        aria-label="Test tabs"
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-l3v1a8-MuiButtonBase-root-MuiTab-root"
          data-testid="tab-control-tabs-c356cfd6-5a7a-4471-9ef3-4c1932251bcc"
          id="tab-unique-id-1"
          role="tab"
          tabindex="0"
          type="button"
        >
          Test Label 1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-l3v1a8-MuiButtonBase-root-MuiTab-root"
          data-testid="tab-control-tabs-c356cfd6-5a7a-4471-9ef3-4c1932251bcc"
          id="tab-unique-id-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Test Label 2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="MuiTabs-indicator css-1ktk2h2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      >
        <span
          class="MuiTabs-indicatorSpan"
        />
      </span>
    </div>
  </div>
  <div
    aria-labelledby="tab-unique-id-1"
    class="MuiBox-root css-1j1uas5"
    data-testid="tab-control-panel-c356cfd6-5a7a-4471-9ef3-4c1932251bcc"
    id="tabpanel-unique-id-1"
    role="tabpanel"
  >
    <div
      style="opacity: 0.2;"
    >
      <div>
        Test Panel 1
      </div>
    </div>
  </div>
</div>
`;
